[CAMEL-17391] Update elasticsearch-rest documentation

assimbly opened the pull request:
- [CAMEL-17391](https://issues.apache.org/jira/browse/CAMEL-17391).
- Based on the solution provided in the Jira I updated the documentation for Elasticsearch. I created a specific paragraph in the adoc. I also want a reference from the options section, because that it where users will look first. 
- Note: This is first time I edited the option section, I assumed I needed to update the json file that will generate this section. If this is incorrect please point to the correct file to update options and I will update this PR.

oscerd reviewed (changes requested):
You need to edit the javadoc in the configuration java class and not directly the json

assimbly commented:
I reverted the change in the generated file and added the comment to comments in component/configuration Java class to be used for Javadoc. If there need to be any other file update please let me know.
